common/src/main/java/com/sequenceiq/cloudbreak/quartz/metric/JobMetricsListener.java

```diff
@@ -46,7 +53,34 @@ public void jobWasExecuted(JobExecutionContext context, JobExecutionException jo
                 jobException == null ? "without exception" : jobException.getMessage());
         QuartzMetricType metricType = jobException == null ? QuartzMetricType.JOB_FINISHED : QuartzMetricType.JOB_FAILED;
         Duration jobRunTime = Duration.ofMillis(context.getJobRunTime());
+        logJobDurationIfCritical(context);
         metricService.recordTimerMetric(metricType, jobRunTime,
                 JOB_GROUP.name(), jobKey.getGroup());
     }
+
+    private void logJobDurationIfCritical(JobExecutionContext context) {
+        long jobRunTime = context.getJobRunTime();
+        if (isJobDurationIsCritical(jobRunTime)) {
+            LOGGER.warn("Critical quartz job ({}), requestId: {}, duration: {}ms, group: {}, class: {}",
```

Comment:
shouldn't we print jobKey.getName()?

Reply:
Most of the time that will be a primary key from the database. I don't think it would give any more help to see it. `requestId` is there, based on that you can investigate the job.